runtime/mount/agent_config/system_prompt.md

```diff
@@ -854,26 +858,30 @@ If assay platform is unclear from data, ask user which platform generated the da
 
 **Turn Actions:**
 
-1. Create plan with steps: "Load data", "Run QC", "Visualize metrics"
-2. Call `submit_response` with plan and `continue: true`
+1. Call `update_plan` to create new plan
+2. Call `submit_response` with `continue: true`
 
 ```python
-submit_response(
+update_plan(
     plan=[
         {"id": "load", "description": "Load spatial data", "status": "todo"},
         {"id": "qc", "description": "Run quality control", "status": "todo"},
         {"id": "viz", "description": "Visualize QC metrics", "status": "todo"}
     ],
     plan_diff=[
-        {"action": "add", "id": "load", "description": "Load spatial data"},
-        {"action": "add", "id": "qc", "description": "Run quality control"},
-        {"action": "add", "id": "viz", "description": "Visualize QC metrics"}
+        {"id": "load", "action": "add"},
+        {"id": "qc", "action": "add"},
+        {"id": "viz", "action": "add"}
     ],
+    plan_update_overview="Created analysis plan with 3 steps."
+)
```

Comment:
It could just re-emit the whole plan vs having a specific update structure. Might improve performance not sure.